redis/cache.py

```diff
@@ -0,0 +1,95 @@
+import random
+import time
+from collections import OrderedDict, defaultdict
+
+
+class _Cache:
+    def __init__(self, max_size: int, ttl: int, eviction_policy: str, **kwargs):
+        self.max_size = max_size
+        self.ttl = ttl
+        self.eviction_policy = eviction_policy
+        self.cache = OrderedDict()
+        self.key_commands_map = defaultdict(set)
+        self.commands_ttl_list = []
+
+    def set(self, command, response, keys_in_command):
+        if len(self.cache) >= self.max_size:
+            self._evict()
+        self.cache[command] = {
+            "response": response,
+            "keys": keys_in_command,
+            "created_time": time.monotonic(),
+        }
+        if self.eviction_policy == "lfu":
+            self.cache[command]["access_count"] = 0
+        self._update_key_commands_map(keys_in_command, command)
+        self.commands_ttl_list.append(command)
+
+    def get(self, command):
+        if command in self.cache:
+            if self._is_expired(command):
+                del self.cache[command]
+                keys_in_command = self.cache[command]["keys"]
+                self._del_key_commands_map(keys_in_command, command)
```

Comment:
self.cache[command].get("keys") for future thread-safe use, and not worrying about `IndexError`